src/Transports.AspNetCore/GraphQLHttpMiddleware.cs

```diff
@@ -46,58 +220,59 @@ public virtual async Task InvokeAsync(HttpContext context, RequestDelegate next)
         var httpRequest = context.Request;
         var httpResponse = context.Response;
 
-        var cancellationToken = GetCancellationToken(context);
-
         // GraphQL HTTP only supports GET and POST methods
         bool isGet = HttpMethods.IsGet(httpRequest.Method);
         bool isPost = HttpMethods.IsPost(httpRequest.Method);
-        if (!isGet && !isPost)
+        if (isGet && !Options.HandleGet || isPost && !Options.HandlePost || !isGet && !isPost)
         {
-            httpResponse.Headers["Allow"] = "GET, POST";
-            await HandleInvalidHttpMethodErrorAsync(context);
+            await HandleInvalidHttpMethodErrorAsync(context, _next);
             return;
         }
 
+        // Authenticate request if necessary
+        if (await HandleAuthorizeAsync(context, _next))
+            return;
+
         // Parse POST body
-        GraphQLRequest bodyGQLRequest = null;
-        IList<GraphQLRequest> bodyGQLBatchRequest = null;
+        GraphQLRequest? bodyGQLRequest = null;
+        IList<GraphQLRequest?>? bodyGQLBatchRequest = null;
         if (isPost)
         {
             if (!MediaTypeHeaderValue.TryParse(httpRequest.ContentType, out var mediaTypeHeader))
             {
-                await HandleContentTypeCouldNotBeParsedErrorAsync(context);
+                await HandleContentTypeCouldNotBeParsedErrorAsync(context, _next);
                 return;
             }
 
             switch (mediaTypeHeader.MediaType)
             {
-                case MediaType.JSON:
```

Comment:
What about `MediaType` static class?

Reply:
I will delete it.

Reply:
Keep in mind that media type support will immediately change in a subsequent PR so that support for "application/graphql+json" is supported.

Reply:
(already done in GraphQL.AspNetCore3)

Reply:
https://github.com/Shane32/GraphQL.AspNetCore3/pull/20/files#diff-7fd8ff58f2234aa7e6cdba5e5893153e2a1ea6647673e4cd7f947c503757e9bb